client/authority-discovery/src/error.rs

```diff
@@ -60,7 +58,7 @@ pub enum Error {
 	ParsingLibp2pIdentity(#[from] libp2p::identity::error::DecodingError),
 
 	#[error("Failed to sign using a specific public key.")]
-	MissingSignature(CryptoTypePublicPair),
```

Comment:
1. If this is going to be a non-newtype `Vec<u8>` then it'd be nice to convert it to a named field so that it's obvious what it is.
2. Should we maybe print out the key in the error message? (Since it's a public key if I'm reading it right.)

Reply:
done.

For the error I maintained the format used for consensus stuff https://github.com/paritytech/substrate/blob/988f6add941b57ceb2af00ab815b3727f03fc1fb/primitives/consensus/common/src/error.rs#L51-L52

I also differentiated the error that was used in case of failure when signing using the network key (a completely different error).